embedded-graphics/src/primitives/arc/linear_equation.rs

```diff
@@ -0,0 +1,48 @@
+use crate::geometry::{Angle, Point, Real, Trigonometry};
+
+/// Define one side of a line
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub(in crate::primitives) enum LineSide {
```

Comment:
If this type is never exported in another public module, this would be inaccessible from the outside, even if it were declared as `pub` here. Should we stick to `pub` for these cases (not only for this type)? I'm not sure if there is any preferred way of handling this.

Reply:
I'm not entirely sure what you mean. Ideally, everything that doesn't need to be `pub` isn't, so there might be some tightening we can do around that.

In this specific case, `LineSide` is also used in the adjacent `plane_sector` module so can't remain private. I could move it into `arc/mod.rs` which would allow it to be private but also used in both modules which might be a better solution.

Reply:
> I'm not entirely sure what you mean. Ideally, everything that doesn't need to be `pub` isn't, so there might be some tightening we can do around that.

Example:
```rust
// in arc/linear_equation.rs
pub(in crate::primitives) struct LinearEquation { ... }

// in arc/mod.rs
pub(in crate::primitives) use linear_equation::LinearEquation;
```
Do we need to specify `in crate::primitives` in both cases? Because if it isn't exported in `mod.rs` it wouldn't be visible outside `primitives::arc` anyway. Reducing the number of specific  `in ...` scopes should make it easier to refactor code.

Reply:
Thanks for explaining, yeah that's not very clean. Shall I keep the `pub(in crate::primitives)` on the item, or on the `use`? I'm leaning towards the item but you might have a good reason to put it on the `use`.

Reply:
I think it needs to be on the `use`, because a simple `pub use ...` would expand the scope and probably cause a compiler error, because the exported type is only visible in the `primitives` module.